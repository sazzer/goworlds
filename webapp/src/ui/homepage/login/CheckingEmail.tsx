import React, {FunctionComponent} from 'react';
import {Form} from "semantic-ui-react";
import {useTranslation} from "react-i18next";

/** The props that the CheckingEmail area needs */
type CheckingEmailProps = {
    email: string,
};

/**
 * The CheckingEmail area
 * @constructor
 */
export const CheckingEmail: FunctionComponent<CheckingEmailProps> = ({email}) => {
    const { t } = useTranslation();

    return (
        <Form loading>
            <Form.Field>
                <label>
                    {t('loginArea.email.label')}
                </label>
                <input placeholder={t('loginArea.email.placeholder')}
                       name="email"
                       type="text"
                       value={email}/>
            </Form.Field>
        </Form>
    );
};
