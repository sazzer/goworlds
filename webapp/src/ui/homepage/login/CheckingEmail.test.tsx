import React from 'react';
import {shallow} from 'enzyme';
import {CheckingEmail} from "./CheckingEmail";

/** Set up the component to test */
function setup(email: string = 'graham@grahamcox.co.uk') {
    const element = shallow(<CheckingEmail email={email}/>);


    return {
        element,
    };
}

it('renders without crashing', () => {
    const {element} = setup();

    expect(element).toMatchSnapshot();
});
