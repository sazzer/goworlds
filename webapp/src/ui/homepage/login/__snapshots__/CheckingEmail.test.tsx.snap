// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Form
  as="form"
  loading={true}
>
  <FormField>
    <label>
      Email Address
    </label>
    <input
      name="email"
      placeholder="Enter Email Address"
      readOnly={true}
      type="text"
      value="graham@grahamcox.co.uk"
    />
  </FormField>
</Form>
`;
